### PR TITLE
feature(docs): Upgrade `docs`

### PR DESCRIPTION
**Pull request by 0xTxbi**
# Description
This PR introduces an upgrade to our `docs` by migrating to TypeScript and updating dependencies. It also adds a new plugin to enhance API documentation support.

# Issues
Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — APPROVED** [review]
I see we lost the redirects? These are important for SEO purposes. Any way to keep them?

File: docs/docusaurus.config.ts [hunk 1]
```diff
@@ -1,208 +1,34 @@
...
-
-/** @type {import('@docusaurus/core').Config} */
-const config = {
+const config: Config = {
   title: 'Unlock Protocol',
   tagline:
     'Unlock is a membership protocol, built on a blockchain. It enables creators to monetize their content or software without relying on a middleman. It lets consumers manage all of their subscriptions in a consistent way, as well as earn discounts when they share the best content and applications they use.',
   url: 'https://docs.unlock-protocol.com',
   baseUrl: '/',
-  onBrokenLinks: 'throw',
```
Comment (julien51) [inline docs/docusaurus.config.ts:13]: I think that is important!

Comment (0xTxbi) [reply docs/docusaurus.config.ts:13]: got it!

File: docs/docusaurus.config.ts [hunk 2]
```diff
@@ -1,208 +1,34 @@
...
+  onBrokenLinks: 'log',
```
Comment (julien51) [inline docs/docusaurus.config.ts:10]:
```suggestion
  onBrokenLinks: 'throw',
```

File: docs/docusaurus.config.ts [hunk 3]
```diff
@@ -14,195 +11,24 @@ const config = {
...
-          {
-            from: '/governance/grants-bounties-and-matchings',
-            to: '/governance/grants-bounties/',
-          },
-        ],
-      },
-    ],
-  ],
+  organizationName: 'unlock-protocol',
+  projectName: 'docs',
```
Comment (julien51) [inline docs/docusaurus.config.ts:15]:
```suggestion
  plugins: [
    'docusaurus-node-polyfills',
    [
      require.resolve('docusaurus-gtm-plugin'),
      {
        id: 'GTM-PRCCFV9',
      },
    ],
    [
      '@docusaurus/plugin-client-redirects',
      {
        redirects: [
          {
            to: '/tools/sign-in-with-ethereum/',
            from: '/unlock/developers/sign-in-with-ethereum',
          },
          {
            to: '/',
            from: '/unlock',
          },
          {
            to: '/',
            from: '/basics',
          },
          {
            to: '/tools/locksmith/webhooks',
            from: '/unlock/tools/locksmith/webhooks',
          },
          {
            to: '/governance/roadmap',
            from: '/unlock/governance/roadmap',
          },
          {
            to: '/tools/sign-in-with-ethereum/unlock-accounts',
            from: [
              '/basics/new-to-unlock/unlock-accounts',
              '/basics/unlock-accounts',
            ],
          },
          {
            to: '/tools/sign-in-with-ethereum/',
            from: '/tools/paywall/sign-in-with-ethereum',
          },
          {
            to: '/getting-started/new-to-web3/ethereum-architecture',
            from: '/new-to-web3/ethereum-architecture',
          },
          {
            to: '/getting-started/new-to-web3/',
            from: '/new-to-web3/',
          },
          {
            to: '/getting-started/new-to-web3/using-etherscan',
            from: '/new-to-web3/using-etherscan',
          },
          {
            to: '/getting-started/new-to-web3/what-is-a-crypto-wallet',
            from: '/new-to-web3/what-is-a-crypto-wallet-a-guide-for-developers',
          },
          {
            to: '/getting-started/new-to-web3/what-is-a-decentralised-application-aka-dapp',
            from: '/new-to-web3/what-is-a-decentralised-application-aka-dapp',
          },
          {
            to: '/getting-started/new-to-web3/what-is-a-smart-contract',
            from: '/new-to-web3/what-is-a-smart-contract',
          },
          {
            to: '/getting-started/new-to-web3/what-is-an-nft',
            from: '/new-to-web3/what-is-an-nft',
          },
          {
            to: '/getting-started/our-code-of-conduct',
            from: '/our-code-of-conduct',
          },
          {
            to: '/getting-started/what-is-unlock/litepaper',
            from: '/litepaper',
          },
          {
            to: '/tutorials/back-end/backend-locking-with-express.js',
            from: '/tutorials/backend-locking-with-express.js',
          },
          {
            to: '/tutorials/front-end/locking-page',
            from: [
              '/tutorials/ad-free-experience',
              '/tutorials/front-end/ad-free-experience',
              '/tutorials/front-end/locking-media-content',
              '/tutorials/locking-media-content',
            ],
          },
          {
            to: '/tutorials/front-end/react-example',
            from: '/tutorials/react-example',
          },
          {
            to: '/tutorials/misc/using-subgraphs',
            from: '/tutorials/using-subgraphs',
          },
          {
            to: '/tutorials/smart-contracts/ethers',
            from: '/tutorials/ethers',
          },
          {
            to: '/tutorials/smart-contracts/hooks/the-key-purchase-hook',
            from: [
              '/tutorials/smart-contracts/the-key-purchase-hook',
              '/tutorials/the-key-purchase-hook',
            ],
          },
          {
            to: '/tutorials/smart-contracts/hooks/using-an-existing-nft-contract',
            from: [
              '/tutorials/smart-contracts/using-an-existing-nft-contract',
              '/tutorials/using-an-existing-nft-contract',
            ],
          },
          {
            to: '/tutorials/smart-contracts/using-unlock-in-other-contracts',
            from: '/tutorials/using-unlock-in-other-contracts',
          },
          {
            to: '/tutorials/smart-contracts/deploying-locally',
            from: '/core-protocol/deploying-locally/',
          },
          {
            to: '/tools/unlock.js',
            from: '/developers/unlock.js',
          },
          {
            to: '/tools/subgraph/entities',
            from: '/tools/entities',
          },
          {
            to: '/tools/subgraph/queries',
            from: '/tools/queries',
          },
          {
            to: '/core-protocol/public-lock/deploying-locks',
            from: '/basics/deploying-a-lock',
          },
          {
            to: '/tools/checkout/configuration',
            from: '/tools/paywall/configuring-checkout',
          },
          {
            to: '/governance/grants-bounties/',
            from: '/governance/grants-bounties-and-matchings',
          },
        ],
      },
    ],
    [
      'docusaurus-plugin-openapi-docs',
      {
        id: 'openapi',
        docsPluginId: 'classic',
        config: {
          locksmith: {
            specPath: '../packages/unlock-js/openapi.yml',
            outputDir: 'docs/api/locksmith',
            sidebarOptions: {
              groupPathsBy: 'tag',
              categoryLinkSource: 'tag',
            },
          },
        },
      },
    ],
  ],
```

**Review by julien51 — APPROVED** [review]
ok, LGTM

**Pull request merged**